app/services/nomisma.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# Class for generating a nomisma RDF file from numismatics coins.
+# Example use: Nomisma.new(output_path: "tmp/princeton-nomisma-2025-11-30.rdf").generate
+class Nomisma
+  attr_reader :graph, :logger, :output_path
+  def initialize(output_path:, logger: Logger.new(STDOUT))
+    @graph = RDF::Graph.new
+    @logger = logger
+    @output_path = output_path
+  end
+
+  def generate
+    coins.sort_by(&:title).each do |coin|
+      logger.info("Processing: #{coin.title}")
+      add_coin_to_graph(coin.decorate)
+    end
+
+    write_xml
+  end
+
+  private
+
+    def coins
+      resources = query_service.find_all_of_model(model: Numismatics::Coin)
+      resources.select { |r| r.state == ["complete"] }
+    end
+
+    def add_coin_to_graph(coin)
+      issue = coin.decorated_parent
+      nmo = RDF::Vocabulary.new("http://nomisma.org/ontology#")
+      void = RDF::Vocabulary.new("http://rdfs.org/ns/void#")
+      dc = RDF::Vocab::DC
+      coin_element = RDF::URI.new(catalog_link(coin))
+
+      graph << RDF::Statement(coin_element, RDF.type, nmo.NumismaticObject)
+      graph << RDF::Statement(coin_element, dc.title, RDF::Literal.new(title(coin)))
+      graph << RDF::Statement(coin_element, dc.identifier, RDF::Literal.new(coin.orangelight_id))
+      graph << RDF::Statement(coin_element, nmo.ObjectType, RDF::Literal.new("coin"))
+      graph << RDF::Statement(coin_element, void.inDataset, RDF::URI.new(numismatics_collection_link))
+      graph << RDF::Statement(coin_element, nmo.hasDiameter, RDF::Literal.new(coin.size, datatype: RDF::XSD.decimal)) if coin.size
+      graph << RDF::Statement(coin_element, nmo.hasWeight, RDF::Literal.new(coin.weight, datatype: RDF::XSD.decimal)) if coin.weight
+      graph << RDF::Statement(coin_element, nmo.hasAxis, RDF::Literal.new(coin.die_axis.first, datatype: RDF::XSD.decimal)) if coin.die_axis.present?
+      graph << RDF::Statement(coin_element, nmo.hasMaterial, RDF::Literal.new(issue.metal.first)) if issue.metal.present?
+      graph << RDF::Statement(coin_element, nmo.hasTypeSeriesItem, RDF::URI.new(coin.primary_reference_uri)) if coin.primary_reference_uri
```

Comment:
Do we not need to find a URI that's specifically for Nomisma?

Reply:
We do not. Nomisma is the whole corpus. In practive, though, all the URLs are for nomisma linked projects.